### PR TITLE
Self hosted pipelines

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Pull Request

## Overview

Updated Azure Pipelines to use Hosted VM and updated tasks for better output.

### CODE REVIEW

File: azure_pipelines/stages/steps/build-unity-player.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 steps:
   - powershell: |
       Write-Host "Starting ${{ parameters.target }} Build..."
-      $logPath = "$(LogDirectory)\${{ parameters.target }}.log"
-      $process = Start-Process -FilePath "$(EditorPath)" -ArgumentList "-quit -batchmode -projectPath `"$(System.DefaultWorkingDirectory)\$(project.name)`" -buildTarget ${{ parameters.target }} -executeMethod XRTK.Utilities.Build.UnityPlayerBuildTools.StartCommandLineBuild -logfile `"$logPath`"" -PassThru
+
+      $date = Get-Date -Format "yyyyMMddTHHmmss"
+      $logPath = "$(LogDirectory)\Build-${{ parameters.target }}-$date.log"
+      $process = Start-Process -FilePath "$(EditorPath)" -ArgumentList "-quit -batchmode -projectPath `"$(System.DefaultWorkingDirectory)\$(project.name)`" -buildTarget ${{ parameters.target }} -executeMethod Emerge.Core.Editor.BuildUtilities.StartCommandLineBuild -logfile `"$logPath`"" -PassThru
```
Comment (StephenHodgson) [inline azure_pipelines/stages/steps/build-unity-player.yml:7]:
```suggestion
      $process = Start-Process -FilePath "$(EditorPath)" -ArgumentList "-quit -batchmode -projectPath `"$(System.DefaultWorkingDirectory)\$(project.name)`" -buildTarget ${{ parameters.target }} -executeMethod XRTK.Utilities.Build.UnityPlayerBuildTools.StartCommandLineBuild -logfile `"$logPath`"" -PassThru
```

File: azure_pipelines/stages/steps/run-unity-tests.yml [hunk 2]
```diff
@@ -35,7 +39,7 @@ steps:
     displayName: 'Publish ${{ parameters.target }} Test Results'
     inputs:
       testResultsFormat: NUnit
-      testResultsFiles: '*test?.xml'
+      testResultsFiles: 'Tests-*?.xml'
```
Comment (StephenHodgson) [inline azure_pipelines/stages/steps/run-unity-tests.yml:42]:
```suggestion
      testResultsFiles: 'Tests-*.xml'
```

File: azure_pipelines/common.yml [hunk 3]
```diff
@@ -4,9 +4,9 @@ parameters:
     default:
       # TODO: Only use XRTK defined platforms for build and install the correct modules and switch to the correct targets as needed
       windows:
-        image: windows-latest
-        # Unity Hub Modules for Windows Editor: android, ios, appletv linux, mac-mono universal-windows-platform, vuforia-ar, webgl, windows-il2cpp, facebook-games, lumin, android-sdk-ndk-tools
-        modules: '"windows-il2cpp", "universal-windows-platform", "android"'
+        agent: xrtk-windows-agent
```
Comment (FejZa) [inline azure_pipelines/common.yml:7]: Can we specify a pool here instead of a specific agent?

**Review by FejZa — DISMISSED** [review]

File: azure_pipelines/common.yml (hunk 3 above)
Comment (StephenHodgson) [reply azure_pipelines/common.yml:7]: I'll make the change and find out

**Pull request merged**